src/ncp/posix/infra_if.cpp

```diff
@@ -376,6 +399,75 @@ bool InfraIf::HasLinkLocalAddress(const std::vector<Ip6Address> &aAddrs)
     return hasLla;
 }
 
+void InfraIf::ReceiveIcmp6Message(void)
+{
+    otbrError error = OTBR_ERROR_NONE;
+    uint8_t   buffer[1500];
+    uint16_t  bufferLength;
+
+    ssize_t         rval;
+    struct msghdr   msg;
+    struct iovec    bufp;
+    char            cmsgbuf[128];
+    struct cmsghdr *cmh;
+    uint32_t        ifIndex  = 0;
+    int             hopLimit = -1;
+
+    struct sockaddr_in6 srcAddr;
+    struct in6_addr     dstAddr;
+
+    memset(&srcAddr, 0, sizeof(srcAddr));
+    memset(&dstAddr, 0, sizeof(dstAddr));
+
+    bufp.iov_base      = buffer;
+    bufp.iov_len       = sizeof(buffer);
+    msg.msg_iov        = &bufp;
+    msg.msg_iovlen     = 1;
+    msg.msg_name       = &srcAddr;
+    msg.msg_namelen    = sizeof(srcAddr);
+    msg.msg_control    = cmsgbuf;
+    msg.msg_controllen = sizeof(cmsgbuf);
+
+    rval = recvmsg(mInfraIfIcmp6Socket, &msg, 0);
+    if (rval < 0)
+    {
+        otbrLogWarning("Failed to receive ICMPv6 message: %s", strerror(errno));
+        ExitNow(error = OTBR_ERROR_DROPPED);
+    }
+
```

Comment:
```suggestion

    VerifyOrExit(ifIndex == mInfraIfIndex, error = OTBR_ERROR_DROPPED);

    // We currently accept only RA & RS messages for the Border Router and it requires that
    // the hoplimit must be 255 and the source address must be a link-local address.
    VerifyOrExit(IN6_IS_ADDR_LINKLOCAL(&srcAddr.sin6_addr), error = OTBR_ERROR_DROPPED);
```

Reply:
`ifIndex` is obtained later in the parsing so I don't think it can be moved here. Thoughts?